src/vs/base/common/fuzzyScorer.ts

```diff
@@ -116,7 +103,17 @@ function doScoreFuzzy(query: string, queryLower: string, queryLength: number, ta
 			// We have a score and its equal or larger than the left score
 			// Match: sequence continues growing from previous diag value
 			// Score: increases by diag score value
-			if (score && diagScore + score >= leftScore) {
+			const isValidScore = score && diagScore + score >= leftScore;
+			if (isValidScore && (
+				// We don't need to check if it's contiguous if we allow non-contiguous matches
+				allowNonContiguousMatches ||
+				// We must be looking for a contiguous match.
+				// Looking at an index higher than 0 in the query means we must have already
+				// found out this is contiguous otherwise there wouldn't have been a score
+				queryIndexGtNull ||
+				// lastly check if the query is completely contiguous at this index in the target
+				target.startsWith(query, targetIndex)
```

Comment:
A micro-optimization would be to not check for `startsWith` if query length = 1 because if we have a score, it means the character must match. But not sure its worth it.

Reply:
sounds like the difference is comparing a single character string vs a number... it _is_ faster... but this if statement is already big enough, I think I'd rather leave that optimization out for simplicity